tests/test_cli_compile.py

```diff
@@ -1909,6 +1909,44 @@ def test_upgrade_package_doesnt_remove_annotation(pip_conf, runner):
         )
 
 
+@pytest.mark.parametrize(("num_inputs"), (2, 3, 10))
+def test_many_inputs_includes_all_annotations(pip_conf, runner, num_inputs):
+    """
+    Tests that an entry required by multiple input files is attributed to all of them in the
+    annotation.
+    See: https://github.com/jazzband/pip-tools/issues/1853
+    """
+    in_files = [f"requirements{n:02d}.in" for n in range(1, num_inputs + 1)]
+
+    for in_file in in_files:
+        with open(in_file, "w") as req_in:
+            req_in.write("small-fake-a==0.1\n")
```

Comment:
This would create requirement files in project dir. I'd suggest using the `tmp_path` fixture:

```suggestion
    in_files = [tmp_path  / f"requirements{n:02d}.in" for n in range(num_inputs)]
    for in_file in in_files:
        req_in.write_text("small-fake-a==0.1\n")
```

Reply:
Aha! So there is a `Path` floating around for this. Excellent.